src/automation/private/StrategiesAutomation.ts

```diff
@@ -19,27 +19,34 @@ import type { ChainId } from '../../types/enums';
 
 interface IStrategiesAutomation extends Interfaces.Automation {
   chainId: ChainId,
+  providerFork: Web3,
 }
 
 export default class StrategiesAutomation extends Automation {
   protected chainId: ChainId;
 
   protected web3: Web3;
 
+  protected web3Fork: Web3;
+
   protected subStorageContract: Contract.WithMeta<SubStorage>;
 
+  protected subStorageContractFork: Contract.WithMeta<SubStorage> | null;
+
   constructor(args: IStrategiesAutomation) {
     super();
 
     this.web3 = args.provider;
+    this.web3Fork = args.providerFork;
     this.chainId = args.chainId;
     this.subStorageContract = makeSubStorageContract(this.web3, this.chainId);
+    this.subStorageContractFork = this.web3Fork ? makeSubStorageContract(this.web3Fork, this.chainId) : null;
 
     this.assert();
   }
 
   protected async getEventFromSubStorage(event: string, options?: PastEventOptions) {
-    return getEventsFromContract<SubStorage>(this.subStorageContract, event, options);
+    return getEventsFromContract<SubStorage>(this.subStorageContract, this.subStorageContractFork, event, options);
   }
 
   protected async getStrategiesSubs(subIds: number[]): Promise<StrategyModel.StoredSubDataStructOutputStruct[]> {
```

Comment:
nece github da mi da da stavim komentar, cini mi se da ces imati problem u ovoj funkciji, linija 62, salje se this.web3 sto ce sada uvek biti mainnet rpc, potencijalno mora i ova funkcija da se podeli u 2 multicalla